rules/strict-equality.js

```diff
@@ -0,0 +1,66 @@
+const eqeqeq = require("eslint/lib/rules/eqeqeq");
+const astUtils = require("eslint/lib/rules/utils/ast-utils");
+
+// - Prefer double equals when comparing with undefined
+// - Prefer undefined over null
+// - Prefer triple equals everywhere else
+module.exports = {
+  meta: {
+    type: "suggestion",
+    fixable: "code",
+    schema: [],
+    messages: {
+      unexpected: "Prefer '{{expectedOperator}}' over '{{actualOperator}}'.",
+    },
+  },
+
+  create: function (context) {
+    const tripleEq = eqeqeq.create(context);
+    const sourceCode = context.getSourceCode();
+
+    function isUndefinedLiteral(node) {
+      return node.type === "Identifier" && node.name === "undefined";
+    }
+
+    return {
+      BinaryExpression: function (node) {
+        function report(node, loc) {
+          let expectedOp = node.operator.substring(0, 2);
+
+          context.report({
+            node,
+            loc,
+            message: `Prefer 'x ${expectedOp} undefined' to catch both null and undefined`,
+          });
+        }
+
+        // If either side is undefined, prefer double equals
+        if (isUndefinedLiteral(node.left) || isUndefinedLiteral(node.right)) {
+          if (node.operator === "===" || node.operator === "!==") {
+            const operatorToken = sourceCode.getFirstTokenBetween(
+              node.left,
+              node.right,
+              (token) => token.value === node.operator
+            );
+
+            report(node, operatorToken.loc);
+          }
+        }
+
+        // If either side is null, prefer undefined
```

Comment:
This seems out of place if the plugin doesn’t include our banning of null everywhere. Did that make it into a rule somewhere? I don’t see it…

Reply:
Yeah, was planning to raise this before landing this plugin in studio. A related question is whether we want to require double equals for undefined, or whether we only suggest it. The current behavior is to ban `undefined === foo` but not `foo === undefined`, which is a bit confusing to me.

Reply:
Side note - I would be perfectly happy keeping the banning of null in this shared config, since it's part of our style guide. However, the `ReactNull` solution isn't self contained, it requires webpack/jest/etc configuration. So that's why I left it out of my first pass.

Maybe we could enable it as a warning and let projects decide what to do.